Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -342,52 +348,69 @@ enum _FileOperations {
                     return
                 }
             }
-        }
 
-        var stack = [path]
-        while let directory = stack.popLast() {
-            guard filemanager?._shouldRemoveItemAtPath(directory) ?? true else { continue }
-            try directory.withNTPathRepresentation {
-                if RemoveDirectoryW($0) { return }
-                let dwError: DWORD = GetLastError()
-                guard dwError == ERROR_DIR_NOT_EMPTY else {
-                    let error = CocoaError.removeFileError(dwError, directory)
-                    guard (filemanager?._shouldProceedAfter(error: error, removingItemAtPath: directory) ?? false) else {
-                        throw error
+            var stack = [path]
+            while let directory = stack.popLast() {
+                try directory.withNTPathRepresentation {
+                    let path = String(decodingCString: $0, as: UTF16.self)
```

Comment:
Would we be able to pick a new name for this variable? It shadows the `path` from the function scope and it was a bit confusing to me seeing this here because I thought that `path` below was the one provided to the function instead of the decoded directory, maybe `standardizedDirectory` or something to denote this is the same "path" as `directory`?

Reply:
Sure, renamed to `ntpath`.